docs: added more info for magic comments (#2707)
webpack/webpack.js.org: src/content/api/module-methods.md

W> This feature relies on [`Promise`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Promise) internally. If you use `import()` with older browsers, remember to shim `Promise` using a polyfill such as [es6-promise](https://github.com/stefanpenner/es6-promise) or [promise-polyfill](https://github.com/taylorhakes/promise-polyfill).

## Magic Comments

> Simple comments to make features work. Just by adding comments to the import we can do things such as name our chunk or select different modes. For a full list of these magic comments see the code below followed by an explanation of what these comments do. 

Reviewer: We had been removing words like "just", "simple", "easy" from the docs not without a reason, i am sorry for another CR but why dont we change it to:
```suggestion
Inline comments to make features work. By adding comments to the import we can do things such as name our chunk or select different modes. For a full list of these magic comments see the code below followed by an explanation of what these comments do. 
```

> The spec for `import` doesn't allow control over the chunk's name or other properties as "chunks" are only a concept within webpack. Luckily webpack allows some special parameters via comments so as to not break the spec. It really is as simple as just adding these comments in order to make these features work:

Reviewer 2: I don't understand this part:
> The spec for `import` doesn't allow control over the chunk's name or other properties as "chunks" are only a concept within webpack.

Do you mean:
"The spec for `import` doesn't allow control over the chunk's name or other properties as "chunks". Chunks is only a webpack concept." ?

Reviewer 2: Also, please remove or rephrase:
> It really is as simple as just adding these comments in order to make these features work:

What is simple for one person it is not for everybody.

Reviewer: It was there already, means there is no "chunks" in spec, but webpack had come up with it :) Though i dont really like the sentence that was added, it does not give any information "simple comments to make features work" where we had already stated that webpack: "allows parameters via comments". literally repeating ourselves with other manner

Author: ok have something that might work better

W> Note that setting `webpackIgnore` to `true` opts out of code splitting.

> `webpackChunkName`: A name for the new chunk. Since webpack 2.6.0, the placeholders `[index]` and `[request]` are supported within the given string to an incremented number or the actual resolved filename respectively. This will cause our separate bundle to be named [my-chunk-name].js instead of just [id].js.

Reviewer: what do you mean by bundle? Doesn't it say your "chunk" will be named x ?

Author: yes your right

> `webpackChunkName`: A name for the new chunk. Since webpack 2.6.0, the placeholders `[index]` and `[request]` are supported within the given string to an incremented number or the actual resolved filename respectively. Adding this comment will cause our separate chunk to be named [my-chunk-name].js instead of just [id].js.

Reviewer: ```suggestion
`webpackChunkName`: A name for the new chunk. Since webpack 2.6.0, the placeholders `[index]` and `[request]` are supported within the given string to an incremented number or the actual resolved filename respectively. Adding this comment will cause our separate chunk to be named [my-chunk-name].js instead of [id].js.
```

- `"eager"`: Generates no extra chunk. All modules are included in the current chunk and no additional network requests are made. A `Promise` is still returned but is already resolved. In contrast to a static import, the module isn't executed until the call to `import()` is made.
- `"weak"`: Tries to load the module if the module function has already been loaded in some other way (i. e. another chunk imported it or a script containing the module was loaded). A `Promise` is still returned but, only successfully resolves if the chunks are already on the client. If the module is not available, the `Promise` is rejected. A network request will never be performed. This is useful for universal rendering when required chunks are always manually served in initial requests (embedded within the page), but not in cases where app navigation will trigger an import not initially served.

> `webpackPrefetch`:  This tells the browser that the resource is probably needed for some navigation in the future. Check out the guide for more information on [how webpackPrefetch works](https://webpack.js.org/guides/code-splitting/#prefetching-preloading-modules).

Reviewer 2: Please remove double space after colon

Reviewer 2: Please use relative URLs: `/guides/code-splitting/#prefetching-preloading-modules`

Reviewer 2: Same comments apply to `webpackPreload`

Reviewer: `This tells the...` "this" is redundant here imo, you are describing `webpackPrefetch` after a colon

Author: done

> `webpackPreload`: Tells the browser that the resource might be needed during the current navigation. Check our the guide for more information on [how webpackPreload works](/guides/code-splitting/#prefetching-preloading-modules).

Reviewer: Typo
```suggestion
`webpackPreload`: Tells the browser that the resource might be needed during the current navigation. Check out the guide for more information on [how webpackPreload works](/guides/code-splitting/#prefetching-preloading-modules).
```

> `webpackPreload`:  This tells the browser that the resource might be needed during the current navigation. Check our the guide for more information on [how webpackPreload works](https://webpack.js.org/guides/code-splitting/#prefetching-preloading-modules).

Reviewer: Same comment as above about `This ...`

Author: got it
